dev-tools/mage/fields.go

```diff
@@ -96,6 +96,11 @@ func GenerateAllInOneFieldsGo() error {
 
 // GenerateFieldsGo generates a .go file containing the fields.yml data.
 func GenerateFieldsGo(fieldsYML, out string) error {
```

Comment:
Could we just change this function to have the BeatName parameter and update the current callers to include that parameter? IMO that's better than adding a new public method since the API surface of this package is already huge.